### PR TITLE
teacher register modify for china

### PR DESCRIPTION
**Pull request by wjllance**
Stop autocomplete for region
Change Chinese label to "school name or training center name"
Only need city and province; switch to province first, then city
Don't need district or country
Phone number required in China
Add WeChat as optional
Default country: China

### CODE REVIEW

File: app/views/core/CreateAccountModal/teacher/SchoolInfoPanel.coffee [hunk 1]
```diff
@@ -84,4 +85,14 @@ SchoolInfoPanel =
   mounted: ->
     $("input[name*='organization']").focus()
 
+    if me.showChinaRegistration()
+      @country = '中国'
```
Comment (nwinter) [inline app/views/core/CreateAccountModal/teacher/SchoolInfoPanel.coffee]: Shouldn't this be `'china'`?

File: app/views/core/CreateAccountModal/teacher/TeacherRolePanel.coffee [hunk 2]
```diff
@@ -9,16 +9,22 @@ TeacherRolePanel = Vue.extend
...
       'referrer'
+      'phoneNumber'
     ])
     return _.assign(formData, {
       showRequired: false
     })
 
+  computed: {
+    validPhoneNumber: ->
+      return forms.validatePhoneNumber(@phoneNumber)
```
Comment (nwinter) [inline app/views/core/CreateAccountModal/teacher/TeacherRolePanel.coffee]: Not sure this worked, I put a phone number of `5` and it went through.

File: app/templates/core/create-account-modal/school-info-panel.jade [hunk 3]
```diff
@@ -11,42 +11,61 @@ div.school-info-panel
...
-      .col-xs-offset-2.col-xs-4
-        .form-group(v-bind:class="{ 'has-error': showRequired && !state }")
-          span.control-label
-            | {{ $t("teachers_quote.state") }}
-            =" "
-            strong(v-if="showRequired && !state") {{ $t("common.required_field") }}
-          input.form-control(name="state", v-model="state")
+    else
+      .row.m-y-2
+        .col-xs-offset-2.col-xs-4
```
[the remark this answers is not in the document]
Comment (wjllance) [reply app/templates/core/create-account-modal/school-info-panel.jade]: can I just keep province/city order the same with  China for other country outside US?

[the remark this answers is not in the document]
Comment (nwinter) [reply app/templates/core/create-account-modal/school-info-panel.jade]: Based on [research here](http://www.columbia.edu/~fdc/postal/) and [here](http://www.bitboost.com/ref/international-address-formats.html), I think we should hide the province/state field if we are not in the USA, Canada, Australia, China, India, Italy, Malaysia, or Mexico. Of those, China can use the `province, city` order and all the others can use the `city, province` order. (This is not an exhaustive list but should be a good-enough start.)

File: app/models/User.coffee [hunk 4]
```diff
@@ -13,6 +13,8 @@ UserLib = {
   broadName: (user) ->
     return '(deleted)' if user.deleted
     name = _.filter([user.firstName, user.lastName]).join(' ')
+    if features?.china
+      name = user.firstName
```
Comment (nwinter) [inline app/models/User.coffee]: Hmm, why do we want to do this? What about users who did split their names into two parts in the previous system?

Comment (wjllance) [reply app/models/User.coffee]: For students, lastName is just a letter, it doesn't make sense in China.

Teachers might split their names, we can tell them to modify it as we have already remove firstName input field on setting and register page

**Pull request merged**